.github/workflows/docker-build-push-github.yaml

```diff
@@ -0,0 +1,61 @@
+# This job builds a docker image and pushes it to Github's registry.
+# It is intended to be called from other workflows or triggered manually.
+# Based on https://docs.github.com/en/actions/publishing-packages/publishing-docker-images#publishing-images-to-github-packages
+
+name: Package OCI
+
+on:
+  # This entry allows calling this workflow from other workflows
+  workflow_call:
+  # This entry allows running this workflow manually
+  workflow_dispatch:
+
+env:
+  REGISTRY: ghcr.io
+  IMAGE_NAME: ${{ github.repository }}
+  # github.head_ref is only populated if this is run for a pull_request
+  IMAGE_TAG: ${{ github.head_ref || github.ref_name }} 
+ 
+jobs:
+  build-and-push-image:
+    runs-on: ubuntu-latest
+    permissions:
+      contents: read
+      packages: write
+
+    steps:
+      - name: Modify env var
```

Comment:
There's an action that already takes care of this, and it's the one used below called `metadata`: https://github.com/docker/metadata-action#branch

Reply:
The metadata defines the value, but this step changes the image tag so that it doesn't contain any invalid characters like slashes. You can think of this step as pre-processing for the metadata step.

Reply:
The action already does that , it's shown in the link referenced above.

Reply:
| Event               | Ref                           | Output           |
|---------------------|-------------------------------|------------------|
| `pull_request`      | `refs/pull/2/merge`           | `pr-2`           |
| `push`              | `refs/heads/master`           | `master`         |
| `push`              | `refs/heads/my/branch`        | `my-branch`      |
| `push tag`          | `refs/tags/v1.2.3`            | `v1.2.3`         |
| `push tag`          | `refs/tags/v2.0.8-beta.67`    | `v2.0.8-beta.67` |
| `workflow_dispatch` | `refs/heads/master`           | `master`         |

Reply:
Look how `my/branch` becomes `my-branch`.

Reply:
Oh, nice. I looked at the docs for that action and evidently missed that. This is actually great, thanks.

Reply:
This step has been removed in favor of `metadata-action` configuration options. Thanks for bringing this up.